Sources/GRPC/Payload.swift

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright YEARS, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import NIO
+import SwiftProtobuf
+
+public protocol Payload {
+    init(serializedByteBuffer: inout NIO.ByteBuffer) throws
+    func serialize(into buffer: inout NIO.ByteBuffer) throws
+}
```

Comment:
We'll need some documentation here. It might also be worth actually naming this `GRPCPayload` instead, what do you think?

Reply:
Yeah that sounds good! `typealise GRPCPayload = NIO.ByteBuffer`

Reply:
Sorry, I meant rename the protocol `Payload` to `GRPCPayload`, not a type alias for `ByteBuffer`

Reply:
Makes sense